executor/src/witgen/generator.rs

```diff
@@ -31,12 +38,42 @@ pub struct Generator<'a, T: FieldElement> {
 impl<'a, T: FieldElement> Machine<'a, T> for Generator<'a, T> {
     fn process_plookup<Q: QueryCallback<T>>(
         &mut self,
-        _mutable_state: &mut MutableState<'a, '_, T, Q>,
+        mutable_state: &mut MutableState<'a, '_, T, Q>,
         _kind: IdentityKind,
-        _left: &[AffineExpression<&'a PolynomialReference, T>],
-        _right: &'a SelectedExpressions<Expression<T>>,
+        left: &[AffineExpression<&'a PolynomialReference, T>],
+        right: &'a SelectedExpressions<Expression<T>>,
     ) -> Option<EvalResult<'a, T>> {
-        unimplemented!()
+        (right.selector == self.latch).then(|| {
+            log::trace!("Start processing secondary VM '{}'", self.name());
+            log::trace!("Arguments:");
+            for (r, l) in right.expressions.iter().zip(left) {
+                log::trace!("  {r} = {l}");
+            }
+
+            let first_row = self
+                .data
+                .last()
+                .cloned()
+                .unwrap_or_else(|| self.compute_partial_first_row(mutable_state));
+
+            let outer_query = OuterQuery {
+                left: left.to_vec(),
+                right,
+            };
+            let ProcessResult { eval_value, block } =
+                self.process(first_row, 0, mutable_state, Some(outer_query));
+
+            if eval_value.is_complete() {
+                log::trace!("End processing VM '{}' (successfully)", self.name());
+                // Remove the last row of the previous block, as it is the first row of the current
```

Comment:
I think I asked a similar question in an older PR: Is it fine to just remove instead of checking that the known values coincide?

Reply:
Depends on how much you distrust the `VmProcessor` ;) It never changes a `CellValue::Known`; all updates are applied via `CellValue::update_with()`, which panics if the value is known already.